bleak/__init__.py

```diff
@@ -204,6 +206,55 @@ def set_scanning_filter(self, **kwargs):
         )
         self._backend.set_scanning_filter(**kwargs)
 
+    def __aiter__(self) -> AsyncIterator[Tuple[BLEDevice, AdvertisementData]]:
+        self._backend.new_devices = asyncio.Queue()
+        return self
+
+    async def __anext__(self) -> Tuple[BLEDevice, AdvertisementData]:
+        return await self._backend.new_devices.get()
+
+    @overload
+    def devices(
+        self, timeout: Optional[float] = None, return_adv: Literal[False] = False
+    ) -> AsyncIterator[BLEDevice]:
+        ...
+
+    @overload
+    def devices(
+        self, timeout: Optional[float] = None, return_adv: Literal[True] = True
+    ) -> AsyncIterator[Tuple[BLEDevice, AdvertisementData]]:
+        ...
+
+    async def devices(self, timeout: Optional[float] = None, return_adv: bool = False):
+        """
+        Continuously scan for devices and yield them.
+
+        Args:
+            timeout:
+                The maximum duration, in seconds, to scan for each individual device.
+                This is not a total scan duration. Each invocation of ``next()`` will
+                wait for up to ``timeout`` seconds for a new device to be discovered
+                before stopping the iteration.
+            return_adv:
```

Comment:
I don't think we should make `return_adv` an option, it should always be enabled. The only reason we have it on other methods is for backwards compatibility. Since this is a new method, we don't need it.

Reply:
This was my first implementation, didn't know that this is for backwards compatibility - but I am glad that this is the case :+1: